portal/plugins/gnmatomresponder/master_importer.py

```diff
@@ -174,28 +174,77 @@ def register_pac_xml(self, content):
         record.save()
         return record
 
+    def check_for_old_finished_jobs(self, vs_item_id):
+        from models import ImportJob
+
+        jobs = ImportJob.objects.filter(item_id=vs_item_id).filter(status='FINISHED')
+
+        number = len(jobs)
+
+        if number > 0:
+            return True
+
+        return False
+
+    def check_key(self, key, vs_item_id):
+        from models import ImportJob
+
+        jobs = ImportJob.objects.filter(item_id=vs_item_id).filter(s3_path=key)
+
+        number = len(jobs)
+
+        if number > 0:
+            return True
+
+        return False
+
+    def check_for_processing(self, vs_item_id):
+        from models import ImportJob
+
+        jobs = ImportJob.objects.filter(item_id=vs_item_id).filter(processing=True)
+
+        number = len(jobs)
+
+        if number > 0:
+            return True
+
+        return False
+
     def import_new_item(self, master_item, content, parent=None):
         from models import ImportJob, PacFormXml
         from pac_xml import PacXmlProcessor
         from mock import MagicMock
         if not isinstance(master_item, VSItem) and not isinstance(master_item, MagicMock): raise TypeError #for intellij
         from portal.plugins.kinesisresponder.sentry import inform_sentry
 
-        vs_item_id = master_item.name
+        vs_item_id = master_item.get("itemId")
 
-        try:
-            importjob = ImportJob.objects.get(item_id=vs_item_id)
-        except ImportJob.DoesNotExist:
-            pass
-        else:
-            if importjob.processing == True:
-                logger.info('Data for item {0} already being processed. Aborting.'.format(vs_item_id))
-                inform_sentry('Data for item {0} already being processed. Aborting.'.format(vs_item_id), {
-                    "master_item": master_item,
-                    "content": content.__dict__,
-                    "parent": parent
-                })
-                return
+        if vs_item_id is None:
+            vs_item_id = master_item.name
+
+        old_finished_jobs = self.check_for_old_finished_jobs(vs_item_id)
+
+        old_key = self.check_key(content['s3Key'], vs_item_id)
+
+        if old_finished_jobs is True and old_key is True:
+            logger.info('Data for item {0} already processed. Aborting.'.format(vs_item_id))
```

Comment:
could you make this error message and the one on line 241 different so we can tell from the logs which is being triggered

Reply:
They are different. One has the word 'being' in it.

Reply:
that's not what i meant :).  One should say that there is already a job _in progress_ and the other that this job has already been successfully completed.  A single word is easy to miss when you're scanning hundreds of lines in a logfile